h/static/scripts/annotator/adder.js

```diff
@@ -30,22 +27,88 @@ var ARROW_HEIGHT = 10;
 var ARROW_H_MARGIN = 20;
 
 /**
- * Controller for the 'adder' toolbar which appears next to the selection
+ * Create the DOM structure for the Adder.
+ *
+ * Returns the root DOM node for the adder, which may be in a shadow tree.
+ */
+function createAdderDOM(container) {
+  var element;
+
+  // If the browser supports Shadow DOM, use it to isolate the adder
+  // from the page's CSS
+  if (container.createShadowRoot) {
+    var root = container.createShadowRoot();
+    root.innerHTML = template;
+    element = root.querySelector('.js-adder');
+
+    // Load stylesheets required by adder into shadow DOM element
+    var adderStyles = Array.from(document.styleSheets).map(function (sheet) {
```

Comment:
On a few sites I looked at that have otherwise fairly strict CSP policies, all allowed "unsafe-inline" in the style-src directive. Perhaps we don't need to worry about it that much.